core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -553,7 +553,7 @@ private class LiveRDDDistribution(exec: LiveExecutor) {
   def toApi(): v1.RDDDataDistribution = {
     if (lastUpdate == null) {
       lastUpdate = new v1.RDDDataDistribution(
-        weakIntern(exec.hostPort),
+        weakIntern(if (exec.hostPort != null) exec.hostPort else exec.host),
```

Comment:
host starts out null as well so It seems like we should really check to make sure both hostPort and host aren't null.

There is code in liveEntity that actually looks at hotspot:
>   def hostname: String = if (host != null) host else Utils.parseHostPort(hostPort)._1

hostPort is set in block manager added and host is set in executor added, I guess order of those messages could be messed up or something weird happened where we didn't get both.

Reply:
@tgravescs  `exec.hostPort` or `exec.host` will be only called when there's a corresponding `LiveExecutor`. And `LiveExecutor` is added either by `SparkListenerExecutorAdded` or `onBlockManagerAdded`, and `SparkListenerExecutorAdded` set the `host` and `onBlockManagerAdded` set the `hostPort`. So, `host` and `hostPort` can't be `null` at the same time.